nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -276,6 +298,12 @@ def __init__(
         self.hidden_size_per_attention_head = safe_divide(projection_size, num_attention_heads)
         self.num_attention_heads_per_partition = safe_divide(num_attention_heads, world_size)
 
+        # Headscale
+        if headscale:
+            self.head_scale_tensor = torch.nn.Parameter(
+                torch.ones(1, self.num_attention_heads_per_partition, 1, self.hidden_size_per_attention_head)
```

Comment:
Should the scaling attention be a scalar per attention head and not per hidden unit in each head?

Reply:
in the paper, it seems to be per attention head
![image](https://user-images.githubusercontent.com/43824965/165165605-ed40cb94-9426-4909-831e-b286a7b62148.png)

Reply:
Yeah I missed the "scalar" part here. Since the equation is ambiguous about whether gamma is a scalar or tensor. Will fix it.